simpleclient_spring_boot/src/main/java/io/prometheus/client/spring/EnablePrometheusCollectorRegistration.java

```diff
@@ -0,0 +1,52 @@
+package io.prometheus.client.spring;
+
+import org.springframework.context.annotation.Import;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An {@literal @}Enable style annotation that provides CollectorRegistry related functions.
+ *
+ * <pre><code>
+ *{@literal @}Configuration
+ *{@literal @}EnablePrometheusCollectorRegistration(useDefault = true, register = true)
+ * public class MyConfiguration {
+ *
+ *  {@literal @}Bean
+ *   public Counter counter() {
```

Comment:
Duplicate metrics, incorrect uses of metrics, inappropriate use of custom registries.

The original java client had code like this, it was over complicated and unnecessary.

Direct instrumentation should use the java client in the usual fashion.